src/app/infinitescroll/infinitescroll.directive.ts

```diff
@@ -22,13 +23,12 @@ export class InfiniteScrollDirective implements OnInit {
   pageSize: number = 10;
   lastCheckedHeight = 0;
   previousScrollHeight = 0;
+  private element: HTMLElement = this.elementRef.nativeElement;
 
   constructor(private elementRef: ElementRef) {
   }
 
-  private element: HTMLElement = this.elementRef.nativeElement;
-
-  ngOnInit() {
+  ngAfterViewChecked() {
```

Comment:
The `ngAfterViewChecked` needs the `@ViewChild` to work properly. Why did you switch to this? What are you trying to do with it?

Reply:
I believe, ```ngAfterViewChecked``` is fired once the view is loaded finished mutation, and we need to check the height of the view at that point. The offsetHeight was giving 0 when the child height is auto because when ```ngOnInit``` called, children inside the view is not loaded.

Reply:
'ngAfterViewChecked' is only for components, it is not meant for directives. Please find another way.

Reply:
I think another way to do this is with ``` ngDoCheck ```. Because ngDoCheck is for Components and as well for Directives.